exercises/practice/rational-numbers/src/RationalNumbers.hs

```diff
@@ -0,0 +1,46 @@
+module RationalNumbers
+(Rational,
+ abs,
+ reduce,
+ add,
+ sub,
+ mul,
+ div,
+ exprational,
+ expreal,
+ rational) where
+
+import Prelude hiding (div, abs, Rational)
+
+-- Data definition -------------------------------------------------------------
+data Rational a = Dummy deriving(Eq, Show)
```

Comment:
Do we really want to parameterize this type? We test no types other than `Integer`.

Reply:
But we want to demonstrate this - cf. Complex Numbers.

Reply:
Parameterizing `Complex` was first suggested in https://github.com/exercism/haskell/pull/601#issuecomment-335623224. Having read the discussion, I still don't see much point to it. But given the `Integral` constraints all over the API I do not foresee any problems either, so I'm fine with keeping it parameterized.